src/app/services/mailchain/mailchain.service.ts

```diff
@@ -328,5 +328,22 @@ export class MailchainService {
     }
   }
 
+  /**
+   * Returns the currency on basis of the selected protocol in Mailchain 
+   * @param protocol the cureent protocol
+   */
+  getCurrencyForProtocol(protocol: string) {
```

Comment:
Is it better for the API to return the currency based on the protocol and network? (e.g. `wei` for current Ethereum supported networks, `edg` for Edgeware-mainnnet, `algo` for Algorand)
This reduces the amount of logic required in the front end, although some is still require for UX, for example a `wei` to `eth` conversion for Ethereum.